user/src/main/java/org/the_chance/honeymart/ui/feature/category/CategoriesScreen.kt

```diff
@@ -81,87 +78,84 @@ fun CategoryContent(
+                            style = MaterialTheme.typography.displaySmall,
+                            color = MaterialTheme.colorScheme.onSecondaryContainer,
                             textAlign = TextAlign.Center,
-                            modifier = Modifier.padding(
-                                top = MaterialTheme.dimens.space16,
-                                bottom = MaterialTheme.dimens.space8
-                            )
                         )
-                        Row(
-                            horizontalArrangement = Arrangement.spacedBy(MaterialTheme.dimens.space4)
-                        ) {
-                            Icon(
-                                painter = painterResource(id = R.drawable.map_pointer),
-                                contentDescription = stringResource(R.string.map_pointer),
-                                tint = MaterialTheme.colorScheme.onSecondaryContainer,
-                            )
-                            Text(
-                                text = state.address,
-                                style = MaterialTheme.typography.displaySmall,
-                                color = MaterialTheme.colorScheme.onSecondaryContainer,
-                                textAlign = TextAlign.Center,
-                            )
-                        }
-                        Row(
-                            modifier = Modifier.padding(vertical = MaterialTheme.dimens.space16),
-                            horizontalArrangement = Arrangement.spacedBy(MaterialTheme.dimens.space8)
-                        ) {
-                            CardChip(
-                                text = "${state.categoriesCount} Categories",
-                                icon = painterResource(id = R.drawable.boxes)
-                            )
-                            CardChip(
-                                text = "${state.productsCount} Items",
-                                icon = painterResource(id = R.drawable.box_minimalistic)
-                            )
-                        }
-                        ImageNetwork(
-                            imageUrl = state.imageUrl,
-                            contentDescription = stringResource(id = R.string.category_image),
-                            modifier = Modifier
-                                .height(198.dp)
-                                .clip(shape = RoundedCornerShape(MaterialTheme.dimens.space24))
-                                .fillMaxWidth(),
-                            contentScale = ContentScale.FillBounds
+                    }
+                    Row(
+                        modifier = Modifier.padding(vertical = MaterialTheme.dimens.space16),
+                        horizontalArrangement = Arrangement.spacedBy(MaterialTheme.dimens.space8)
+                    ) {
+                        CardChip(
+                            text = "${state.categoriesCount} Categories",
```

Comment:
This concatenation should be done within the view model, right?

Reply:
@TarekIdrees where in viewmodel you mean? like in uistate model?

Reply:
you can create a state that hold the final concatenation of count and placholder like 300$ for example.

Reply:
@TarekIdrees if you mean a state in the screen uistate as far as i understand it would look like this. is that what you mean?
`) {
    val productsCountState = "$productsCount Products"
    val categoriesCountState = "$categoriesCount Products"
}`

Reply:
Not exactly, you can create a ui state that will be provided by the view model like categoryState and within the view model this state will be provided and then will be used in the screen.